gnucash/gnome-utils/gnc-file.c

```diff
@@ -1102,9 +1103,20 @@ gnc_post_file_open (GtkWindow *parent, const char * filename, gboolean is_readon
         g_free ( message );
     }
 
-    // Fix account color slots being set to 'Not Set', should run once on a book
     qof_event_suspend();
+
+    // Fix account color slots being set to 'Not Set', should run once on a book
     xaccAccountScrubColorNotSet (gnc_get_current_book());
+
+    /* Fix budget signs */
+    budget_scrubbing_outcome = gnc_scrub_budget_signs (new_book);
```

Comment:
`gnc_post_file_open` is already 10x longer than is reasonable so put the new stuff  into a static function `maybe_scrub_budget_signs` that:
1. Checks the feature and whether there's a budget. It should call `gnc_scub_budget_signs` only if necessary.
2. `gnc_scrub_budget_signs` can return a boolean that's TRUE if it actually found something to scrub.
3. Display a dialog box only if `gnc_scrub_budget_signs` that tells the user that the scrub occurred and that they should check their budgets to make sure it did it right. Don't forget to mark the message for translation. 

Call `maybe_scrub_budget` here.

Reply:
Leaving the feature_check and budget_check to `ScrubBudget.c` seems to make more sense to me? See latest.